### PR TITLE
KAFKA-2893: Add a simple non-negative partition seek check

### PR DESCRIPTION
**Pull request by jinxing64**

### CODE REVIEW

**Comment by jinxing64** [comment]
hi, @guozhangwang , do you mind to take a look at this?

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java [hunk 1]
```diff
@@ -1007,6 +1007,9 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      */
     @Override
     public void seek(TopicPartition partition, long offset) {
+        if (offset < 0) {
+            throw new IllegalArgumentException("offset for seek must not be a negative number");
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: "offset for seek" => "seek offset"

**Comment by guozhangwang** [comment]
@ZoneMayor Could you add a test case in KafkaConsumerTest for this scenario as well?

**Comment by jinxing64** [comment]
@guozhangwang thanks a lot for your feedback, I will do that

**Comment by jinxing64** [comment]
@guozhangwang 
Hi Guozhang,  added a unit test for seek offset;
please help chenck again  : )

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java [hunk 2]
```diff
@@ -78,4 +79,24 @@ public void testSubscription() {
...
+        props.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, "testSeekNegative");
+        props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        props.setProperty(ConsumerConfig.METRIC_REPORTER_CLASSES_CONFIG, MockMetricsReporter.class.getName());
+
+        KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<byte[], byte[]>(
+                props, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+        consumer.assign(Arrays.asList(new TopicPartition("nonExistTopic", 0)));
+        consumer.seek(new TopicPartition("nonExistTopic", 0), 1);
+        try {
+            consumer.seek(new TopicPartition("nonExistTopic", 0), -1);
```
Comment (guozhangwang) [inline clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java]: We can use JUnit "expect exception" here. For example in SchemaBuilderTest.testInt64BuilderInvalidDefault.

**Comment by jinxing64** [comment]
Hi ~@guozhangwang  Thanks a lot for your comment, I make it to be expected exception, please help check again ~thanks

**Comment by guozhangwang** [comment]
LGTM. Merged to both branches.

**Pull request closed without merging**